File: server/src/routes/knowledgeBase.ts
import { Router, Request, Response } from 'express';
import { ragflowService } from '../services/ragflow.js';

const router = Router();

// GET / - List all knowledge bases (datasets)
router.get('/', async (_req: Request, res: Response) => {
  try {
    const kbs = await ragflowService.getKnowledgeBases();
    res.json({ code: 0, data: kbs });
  } catch (error: any) {
    console.error('Failed to fetch knowledge bases:', error.message);
    res.status(500).json({ code: 1, message: error.message });
  }
});

export default router;

File: server/app/routers/knowledge_base.py
# -*- coding: utf-8 -*-
"""Knowledge base routes: /api/knowledge-bases (equivalent to original server/src/routes/knowledgeBase.ts)."""
import logging
from fastapi import APIRouter

from ..ragflow import ragflow_service
from ..security import ApiError

logger = logging.getLogger(__name__)

router = APIRouter()


@router.get("")
async def list_knowledge_bases():
    try:
        kbs = await ragflow_service.get_datasets()
        return {"code": 0, "data": kbs}
    except Exception as e:
        logger.exception(f"Failed to fetch knowledge bases: {e}")
        raise ApiError(500, 1, str(e))
